docs/proposals/06-immutable-etcd-backups.md

```diff
@@ -0,0 +1,254 @@
+
+## Alternatives
+
+### Object-Level Immutability Policies vs. Bucket-Level Immutability Policies
+
+An alternative to implementing immutability via bucket-level immutability policies is to use object-level immutability policies. Object-level immutability allows for more granular control over the immutability periods of individual objects within a bucket, whereas bucket-level immutability applies a uniform immutability period to all objects in the bucket.
+
+#### Feasibility Study: Immutable Backups on Cloud Providers
+
+Major cloud storage providers such as Google Cloud Storage (GCS), Amazon S3, and Azure Blob Storage (ABS) support both bucket-level and object-level immutability mechanisms to enforce data immutability.
+
+1. **Bucket-Level Immutability Policies:**
+   - **Applies Uniformly:** Applies a uniform immutability period to all objects within a bucket.
+   - **Immutable Objects:** Once set, objects cannot be modified or deleted until the immutability period expires.
+   - **Simplified Management:** Simplifies management by applying the same policy to all objects.
+
+2. **Object-Level Immutability Policies:**
+   - **Granular Control:** Allows setting immutability periods on a per-object basis.
+   - **Flexible Immutability Durations:** Offers granular control, enabling different immutability durations for individual objects.
+   - **Varying Requirements:** Can accommodate varying immutability requirements for different types of backups.
+
+#### Considerations for Object-Level Immutability
+
+Using object-level immutability provides flexibility in scenarios where certain backups require different immutability periods. However, current limitations and complexities make it less practical for immediate implementation.
+
+- Enabling object-level immutability requires bucket-level immutability to be set first (applicable in S3 and ABS). In GCS, the capability to enable object-level immutability on an existing bucket is not available.
+
+**Advantages:**
+
+- **Granular Control:** Allows setting different immutability periods for different objects.
+- **Efficient Resource Utilization:** Prevents unnecessary extension of immutability for all objects.
+- **Enhanced Flexibility:** Adjust immutability periods as needed.
+
+**Disadvantages:**
+
+- **Provider Limitations:** Enabling object-level immutability on existing buckets is not universally supported.
+- **Increased Complexity:** Requires additional logic in backup processes and tooling.
```

Comment:
What is the increase in complexity as compared to what we are already introducing for bucket level immutability policies?